Sources/GRPCCore/Timeout.swift

```diff
@@ -0,0 +1,226 @@
+  /// See "Timeout" in https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests
+  var wireEncoding: String {
+    "\(amount)\(unit.rawValue)"
+  }
+
+  @usableFromInline
+  var description: String {
+    return self.wireEncoding
+  }
+
+  @usableFromInline
+  init?(decoding value: String) {
+    guard (2 ... 8).contains(value.count) else {
+      return nil
+    }
+
+    if let amount = Int64(value.dropLast()),
+      let unit = Unit(rawValue: value.last!)
+    {
+      self = Self.init(amount: amount, unit: unit)
+    } else {
+      return nil
+    }
+  }
+
+  /// Create a ``Timeout`` from a ``Duration``.
+  ///
+  /// - Important: It's not possible to know with what precision the duration was created: that is,
+  /// it's not possible to know whether `Duration.seconds(value)` or `Duration.milliseconds(value)`
+  /// was used. For this reason, the unit chosen for the ``Timeout`` (and thus the wire encoding) may be
+  /// different from the one originally used to create the ``Duration``. Despite this, we guarantee that
+  /// both durations will be equivalent if there was no loss in precision during the transformation.
+  /// For example, `Duration.hours(123)` will yield a ``Timeout`` with `wireEncoding` equal to
+  /// `"442800S"`, which is in seconds. However, 442800 seconds and 123 hours are equivalent.
+  /// However, you must note that there may be some loss of precision when dealing with transforming
+  /// between units. For example, for very low precisions, such as a duration of only a few attoseconds,
+  /// given the smallest unit we have is whole nanoseconds, we cannot represent it. Same when converting
+  /// for instance, milliseconds to seconds. In these scenarios, we'll round to the closest whole number in
+  /// the target unit.
+  @usableFromInline
+  init(duration: Duration) {
+    let (seconds, attoseconds) = duration.components
+
+    if seconds == 0 {
+      // There is no seconds component, so only pay attention to the attoseconds.
+      // Try converting to nanoseconds first, and continue rounding up if the
+      // max amount of digits is exceeded.
+      let nanoseconds = Int64(round(Double(attoseconds) / 1e+9))
+      self.init(rounding: nanoseconds, unit: .nanoseconds)
+    } else if Self.exceedsDigitLimit(seconds) {
+      // We don't have enough digits to represent this amount in seconds, so
+      // we will have to use minutes or hours.
+      // We can also ignore attoseconds, since we won't have enough precision
+      // anyways to represent the (at most) one second that the attoseconds
+      // component can express.
+      self.init(rounding: seconds, unit: .seconds)
+    } else {
+      // We can't convert seconds to nanoseconds because that would take us
+      // over the 8 digit limit (1 second = 1e+9 nanoseconds).
+      // We can however, try converting to microseconds or milliseconds.
+      let nanoseconds = Int64(Double(attoseconds) / 1e+9)
```

Comment:
nit: we round when doing this conversion on L102 but not here. Let's pick one and be consistent.